client/engine/chainservice/mockchain.go

```diff
@@ -50,6 +59,13 @@ func (mc *MockChain) SubscribeToEvents(a types.Address) <-chan Event {
 // SendTransaction responds to the given tx.
 func (mc *MockChain) SendTransaction(tx protocols.ChainTransaction) {
 	*mc.blockNum++
+	if mc.txListener != nil {
+		// Send to txListener and ignore if the chan is full
```

Comment:
Should we ignore the "channel full" error? Seems like a full channel might lead to behavior that is tricky to debug.